### PR TITLE
fixed support for results other than screenshots

### PR DESCRIPTION
**Pull request by nadvornik**
- support screenshots (png), audio (wav) and text (txt)
- use md5 for all these files
- dropped broken ocr results, use more generic 'text' instead

### CODE REVIEW

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -127,12 +127,17 @@ sub save_base64_png($$$) {
...
+
+    my $ext;
+    $ext = 'txt' if defined $orig_fname && $orig_fname =~ /\.txt$/;
+    $ext = 'wav' if defined $orig_fname && $orig_fname =~ /\.wav$/;
+    $ext //= 'png';
 
     my $prefix = substr($md5, 0, 2);
     $md5 = substr($md5, 2);
-    return ($imagesdir . "/$prefix/$md5.png", $imagesdir . "/$prefix/.thumbs/$md5.png");
+    return ($imagesdir . "/$prefix/$md5.$ext", $imagesdir . "/$prefix/.thumbs/$md5.$ext");
```
Comment (coolo) [inline lib/OpenQA/Utils.pm]: so you're saving wav and txt files in /images ? That doesn't sound proper

**Comment by coolo** [comment]
I'm torn if it makes sense to store the txt file there too. How does the thumbnail of a text file look like?

**Comment by coolo** [comment]
oops, didn't mean to close

**Comment by nadvornik** [comment]
As a thumbnail it shows images/text.png.

For parsed junit file it looks like this:
http://dhcp77.suse.cz/tests/1542

IMHO it is ok for now, it can be improved later.

File: lib/OpenQA/Utils.pm (hunk 1 above)
Comment (nadvornik) [inline lib/OpenQA/Utils.pm]: You are right, OTOH adding new directories would be just a cosmetic change, so I am not sure if it is worth it.

Comment (coolo) [inline lib/OpenQA/Utils.pm]: It's just *WRONG* - that's all. Why not save these in testresults ? they are very likely much much smaller - and the optipng reason does not apply to them.

**Comment by coolo** [comment]
btw: why are there As without result border?

**Comment by nadvornik** [comment]
here is a version without symlinks

As without result border are testsuite failures, they are probably not mapped to openqa results correctly, this part is not finished yet.

File: templates/step/moduleslistthumbnails.html.ep [hunk 2]
```diff
@@ -13,6 +13,10 @@
                     %= link_to url_for('step', stepid => $img_count) => (class => 'no_hover') => begin
                         %= image '/images/audio.svg', alt => $refimg->{audio}, title => $refimg->{audio}, width => $ref_width, height => $ref_height, class=> "resborder_\L$refimg->{result}"
                     % end
+                % } elsif ($refimg->{text}) {
+                    %= link_to url_for('step', stepid => $img_count) => (class => 'no_hover') => begin
+                        %= image '/images/text.png', alt => $refimg->{audio}, title => $refimg->{audio}, width => $ref_width, height => $ref_height, class=> "resborder_\L$refimg->{result}"
+                    % end
```
Comment (coolo) [inline templates/step/moduleslistthumbnails.html.ep]: you're using ->{audio} here - sounds not proper

File: lib/OpenQA/Controller/Step.pm [hunk 3]
```diff
@@ -66,6 +66,10 @@ sub init {
     else {
         my $module_detail = $details->[$testindex - 1];
         $tabmode = 'audio' if ($module_detail->{'audio'});
+        if ($module_detail->{'text'}) {
```
Comment (coolo) [inline lib/OpenQA/Controller/Step.pm]: I know it's the line above used too, but please no extra '' :)

**Comment by nadvornik** [comment]
fixed

**Comment by nadvornik** [comment]
and rebased

**Pull request merged**